packages/grafana-ui/src/components/Text/Text.tsx

```diff
@@ -151,3 +112,83 @@ const getTextStyles = (
     },
   ]);
 };
+
+function TruncatedText({
+  element = 'span',
+  variant,
+  color,
+  weight,
+  italic,
+  textAlignment,
+  children,
+  ...restProps
+}: TextProps) {
+  const styles = useStyles2(
+    useCallback(
+      (theme) => getTextStyles(theme, element, variant, color, weight, true, italic, textAlignment),
+      [color, textAlignment, italic, weight, variant, element]
+    )
+  );
+  const [isOverflowing, setIsOverflowing] = useState(false);
+  const internalRef = useRef<HTMLElement>(null);
+
+  const resizeObserver = useMemo(
+    () =>
+      new ResizeObserver((entries) => {
+        for (const entry of entries) {
+          if (entry.target.clientWidth && entry.target.scrollWidth) {
+            if (entry.target.scrollWidth > entry.target.clientWidth) {
+              setIsOverflowing(true);
+            }
+            if (entry.target.scrollWidth <= entry.target.clientWidth) {
+              setIsOverflowing(false);
+            }
+          }
+        }
+      }),
+    []
+  );
+
+  useEffect(() => {
+    const { current } = internalRef;
+    if (current) {
+      resizeObserver.observe(current);
+    }
+    return () => {
+      resizeObserver.disconnect();
+    };
+  }, [isOverflowing, resizeObserver]);
+
+  const getTooltipText = (children: NonNullable<React.ReactNode>) => {
+    if (typeof children === 'string') {
+      return children;
+    }
+    const html = ReactDomServer.renderToStaticMarkup(<>{children}</>);
+    const getRidOfTags = html.replace(/(<([^>]+)>)/gi, '');
+    return getRidOfTags;
+  };
+
+  if (isOverflowing && element !== 'span') {
+    return (
+      <Tooltip ref={internalRef} content={getTooltipText(children)}>
+        {createElement(element, {
+          ...restProps,
+          style: undefined,
+          className: styles,
+          ref: internalRef,
```

Comment:
According to the original code this should be undefined. Is there a reason why you changed it?